Mage/src/main/java/mage/watchers/Watcher.java

```diff
@@ -75,18 +77,61 @@ public void reset() {
         condition = false;
     }
 
-    protected String getBasicKey(){
+    protected String getBasicKey() {
         return getClass().getSimpleName();
     }
 
     public abstract void watch(GameEvent event, Game game);
 
-    public <T extends Watcher> T copy(){
+    public <T extends Watcher> T copy() {
         try {
-            Constructor<? extends Watcher> constructor = this.getClass().getDeclaredConstructor(getClass());
+            List<?> constructors = Arrays.asList(this.getClass().getConstructors());
+            if (constructors.size() > 1) {
+                logger.error(getClass().getSimpleName() + " has multiple constructors");
+                return null;
+            }
+
+            Constructor<? extends Watcher> constructor = (Constructor<? extends Watcher>) constructors.get(0);
+            if (constructor.getParameterCount() > 0) {
+                logger.error(getClass().getSimpleName() + " constructor has arguments, should be 0 and inject the parameters by setters");
+                return null;
+            }
             constructor.setAccessible(true);
-            return (T) constructor.newInstance(this);
-        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            T watcher = (T) constructor.newInstance();
+            List<Field> allFields = new ArrayList<>();
+            allFields.addAll(Arrays.asList(getClass().getDeclaredFields()));
```

Comment:
You likely would want to do something like this to make sure that even if someone extended another watcher, their fields would also get copied.
```
        Class current = getClass();
        while (current != null) {
            allFields.addAll(/* the declared fields */)
            current = current.getSuperclass();
        }
```

Reply:
You are correct, extending an existing Watcher is not supported. However I do not know if inheritance is a good design principle for those Watchers. 

The ability.newId() line is not in any of the Watchers in this PR, this PR does not cover ability watchers or effects. 

None, but 1 watcher, has a hard-copy. The copy method by reflection is used by default, and that can be overridden.

Reply:
Out of scope then, but I could see making watchers explicitly final then.

Reply:
Agree.